3_harmonize/src/harmonize_sdd.R

```diff
@@ -184,312 +171,887 @@ harmonize_sdd <- function(raw_sdd, p_codes,
+    # the mena value (harmonized_value in this case)
+    mutate(
+      harmonized_value_cv = harmonized_value_sd/harmonized_value
+    ) %>% 
+    ungroup() %>% 
+    select(
+      # no longer needed
+      -harmonized_value_sd) %>% 
+    relocate(
+      c(subgroup_id, harmonized_row_count, harmonized_units,
+        harmonized_value, harmonized_value_cv, lat, lon, datum),
+      .after = misc_flag
+    )
   
-  grouped_equipment_sdd <- grouped_sample_methods_sdd %>%
-    left_join(x = .,
-              y = sdd_equipment_matchup,
-              by = c("SampleCollectionEquipmentName" = "collection_equipment")) %>%
-    filter(equipment_grouping != "unlikely")
+  rm(grouped_sdd)
+  gc()
   
-  # How many records removed due to unlikely sample methods?
+  # Plot harmonized measurements by Tier:
+  
+  # 1. Harmonized values
+  tier_dists <- no_simul_sdd %>% 
+    select(tier, harmonized_value) %>% 
+    mutate(plot_value = harmonized_value+0.001,
+           tier_label = case_when(
+             tier == 0 ~ "Restrictive (Tier 0)",
+             tier == 1 ~ "Narrowed (Tier 1)",
+             tier == 2 ~ "Inclusive (Tier 2)",
+             tier == 3 ~ "Inclusive (Tier 3)"
+           )) %>% 
+    ggplot() +
+    geom_histogram(aes(plot_value)) +
+    facet_wrap(vars(tier_label), scales = "free_y", ncol = 1) +
+    xlab(expression("Harmonized SDD (m, " ~ log[10] ~ "transformed)")) +
+    ylab("Count") + 
+    ggtitle(label = "Distribution of harmonized SDD values by tier", 
+            subtitle = "0.001 added to each value for the purposes of visualization only") +
+    scale_x_log10(label = label_scientific()) +
+    scale_y_continuous(label = label_scientific()) +
+    theme_bw() +
+    theme(strip.text = element_text(size = 7))
+
+    ggsave(filename = "3_harmonize/out/sdd_tier_dists_postagg.png",
+            plot = tier_dists,
+            width = 6, height = 4, units = "in", device = "png")
+  
+  # 2: Harmonized CVs
+  # Count NA CVs in each tier for plotting
+  na_labels <- no_simul_sdd %>% 
+    filter(is.na(harmonized_value_cv)) %>% 
+    count(tier)
+  
+  tier_0 <- filter(na_labels, tier == 0)$n %>% 
+    number(scale_cut = cut_short_scale())
+  
+  tier_1 <- filter(na_labels, tier == 1)$n %>% 
+    number(scale_cut = cut_short_scale())
+  
+  tier_2 <- filter(na_labels, tier == 2)$n %>% 
+    number(scale_cut = cut_short_scale())
+  
+  tier_3 <- filter(na_labels, tier == 3)$n %>% 
+    number(scale_cut = cut_short_scale())
+  
+  tier_cv_dists <- no_simul_sdd %>% 
+    select(tier, harmonized_value_cv) %>% 
+    mutate(plot_value = harmonized_value_cv + 0.001,
+           tier_label = case_when(
+             tier == 0 ~ paste0("Restrictive (Tier 0) NAs removed: ", tier_0),
+             tier == 1 ~ paste0("Narrowed (Tier 1) NAs removed: ", tier_1),
+             tier == 2 ~ paste0("Inclusive (Tier 2) NAs removed: ", tier_2),
+             tier == 3 ~ paste0("Inclusive (Tier 2) NAs removed: ", tier_3)
```

Comment:
```suggestion
             tier == 3 ~ paste0("Inclusive (Tier 3) NAs removed: ", tier_3)
```
Also, see the suggestion about using factors above.